components/Contact.js

```diff
@@ -1,58 +1,171 @@
+// @author William Shi, Charles Liu
+
 import React from 'react'
+import { useState, useEffect } from 'react'
 import Title from './Title'
-import HomeMadeButton from './shared/Button'
 import { Form, Col } from 'react-bootstrap'
+import Flip from 'react-reveal/Flip'
+import emailjs from 'emailjs-com'
 
+import Success from './shared/Success'
 import styles from '../styles/Contact.module.css'
 
-const Contact = () => {
+const Contact = ({ name, email, subject, message }) => {
+  const [validated, setValidated] = useState(false)
+  const [showErrorMessages, setShowErrorMessages] = useState(false)
+
+  const [contactForm, setContactForm] = useState({
+    name: name,
+    email: email,
+    subject: subject,
+    message: message,
+  })
+
+  useEffect(() => {
+    const timeout = setTimeout(() => setValidated(false), 4000)
+  }, [validated])
+
+  const inputChangedHandler = (e, key) => {
+    const updatedContactForm = { ...contactForm }
+
+    const updatedContactEl = { ...updatedContactForm[key] }
+    updatedContactEl.value = e.target.value
+    updatedContactForm[key] = updatedContactEl
+    setContactForm(updatedContactForm)
+  }
+
+  const submitHandler = (e) => {
+    const form = e.target
+    e.preventDefault()
+    e.stopPropagation()
```

Comment:
woAH. ive never seen `stopPropagation` before lol. what does it do? Google tells me it has smtg do to with event bubbling.
do you also need `preventDefault` then?

Reply:
I don't really need the stopPropogation method because the event isn't really bubbling to anything else but I kept it there because I reused a code snippet on EmailJS documentation